### ISSUE TITLE
Questions on cpql

### ISSUE THREAD

**Issue by wadx2019** [cccedric/cpql#1]
Hello, authors of CPQL! The idea of CPQL is very interesting. Thus, I reproduce the experiments on Humanoid-v3 and HalfCheetah-v3 with the code of this repo. I completely followed the instructions in readme and did not tune any hyper-parameters. However, it seems that a large gap exists between the reproduced results and the reported results.  So can you help me run CPQL to achieve the reported performance?

**Comment by cccedric** [comment]
Hi Irvine, 

Thanks for your interest in cpql. Could you please provide me with more information, such as the training curves, seeds, and so on? And what about other online experiments of MuJoCo, including Swimmer-V3, Walker2d-V3, and Ant-V3?

Best,
Yuhui

**Comment by wadx2019** [comment]
[HalfCheetah-v3.txt](https://github.com/cccedric/cpql/files/14771868/HalfCheetah-v3.txt)
[main.txt](https://github.com/cccedric/cpql/files/14772042/main.txt)

Here are the training results and the evaluation results per 10000 iters of HalfCheetah-v3. The only file I modified is the attached main.py(main.txt) and I just makes it output the intermediate results without any further modification. The command line instruction is as your instruction:
```
python main.py --rl_type online --env_name HalfCheetah-v3
```

**Comment by cccedric** [comment]
It's a weird result. I just checked the experiment results for the paper. The seeds selected for HalfCheetah-V3 are 0, 1, 2, 3, 4, 5, and the worst reward is more than 9500. I will check whether the code in the repo is correct in a few days. Also, could you please have a look at other experiments, such as online tasks of DMC and the offline tasks? Please report to me if you get a similar bad result on other tasks. Thanks!

Best,
Yuhui

**Comment by wadx2019** [comment]
Similar results in Hopper-v3 and Humanoid-v3. In one hand, it converges slowly before 500,000 iters. In the other hand, after 500,000 iters, while the reward improved, it is very unsteady.

**Comment by cccedric** [comment]
Apologies for the delayed response. We spent significant time comparing the code in the repo with those for the paper. Eventually, we found that the solution was very easy: 
1. Remove the NormLayer from the critic network. (start from line 50 in model.py)
2. Remove the normalization of the q value when updating 2 Q networks. (start from line 158 in ql_cm.py)

After testing the updated code on HalfCheetah-v3 and Humanoid-v3, we confirmed that the performance matches the description in the paper.

We'll update the repo later today to implement this fix. Thank you for your interest in our work, and we hope you continue to follow our progress.

Best,
Yuhui
